python-checks/src/test/java/org/sonar/python/checks/utils/PythonCheckVerifier.java

```diff
@@ -0,0 +1,229 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks.utils;
+
+import com.google.common.base.Function;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Ordering;
+import com.sonar.sslr.api.AstAndTokenVisitor;
+import com.sonar.sslr.api.Grammar;
+import com.sonar.sslr.api.Token;
+import com.sonar.sslr.api.Trivia;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import org.sonar.python.PythonAstScanner;
+import org.sonar.python.PythonCheck;
+import org.sonar.python.PythonCheck.IssueLocation;
+import org.sonar.python.PythonCheck.PreciseIssue;
+import org.sonar.squidbridge.checks.SquidCheck;
+
+import static org.fest.assertions.Assertions.assertThat;
+import static org.junit.Assert.fail;
+
+public class PythonCheckVerifier extends SquidCheck<Grammar> implements AstAndTokenVisitor {
+
+  private List<TestIssue> expectedIssues = new ArrayList<>();
+
+  public static void verify(File file, PythonCheck check) {
+    PythonCheckVerifier verifier = new PythonCheckVerifier();
+    PythonAstScanner.scanSingleFile(file, check, verifier);
+
+    Iterator<PreciseIssue> actualIssues = getActualIssues(check);
+    List<TestIssue> expectedIssues = verifier.expectedIssues;
+
+    for (TestIssue expected : expectedIssues) {
+      if (actualIssues.hasNext()) {
+        verifyIssue(expected, actualIssues.next());
+      } else {
+        throw new AssertionError("Missing issue at line " + expected.line());
+      }
+    }
+
+    if (actualIssues.hasNext()) {
+      PreciseIssue issue = actualIssues.next();
+      throw new AssertionError("Unexpected issue at line " + line(issue) + ": \"" + issue.primaryLocation().message() + "\"");
+    }
+
+  }
+
+  private static int line(PreciseIssue issue) {
+    return issue.primaryLocation().startLine();
+  }
+
+  private static void verifyIssue(TestIssue expected, PreciseIssue actual) {
+    if (line(actual) > expected.line()) {
+      fail("Missing issue at line " + expected.line());
+    }
+    if (line(actual) < expected.line()) {
+      fail("Unexpected issue at line " + line(actual) + ": \"" + actual.primaryLocation().message() + "\"");
+    }
+    if (expected.message() != null) {
+      assertThat(actual.primaryLocation().message()).as("Bad message at line " + expected.line()).isEqualTo(expected.message());
+    }
+    if (expected.effortToFix() != null) {
+      assertThat(actual.cost()).as("Bad effortToFix at line " + expected.line()).isEqualTo(expected.effortToFix());
+    }
+    if (expected.startColumn() != null) {
+      assertThat(((PreciseIssue) actual).primaryLocation().startLineOffset() + 1).as("Bad start column at line " + expected.line()).isEqualTo(expected.startColumn());
```

Comment:
@vilchik-elena Eclipse tells me that there is an unnecessary cast to PreciseIssue here and on lines 89 and 92.